app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = @user.posts.find(params['post_id'])
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to user_post_path(@user, @post) }
+      else
+        format.html { render :new }
+      end
+    end
+  end
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```

Comment:
- [ ] Good job using `params` in your code base. However,  for best practices reasons, I think it would be better if you make your `comment_params` method as `private` method. You can update it to some thing like this:
```ruby
...
private

  def comment_params
    params.require(:post).permit(:title, :text)
  end
end
```
**I suggest you apply this appoach in other controllers with the same logic**

Reply:
Thank you! I'll implement this.